### PR TITLE
TS Check, Lint, and Format `new-package/` and `vendor/`

### PR DESCRIPTION
**Pull request by lishaduck**
Part 1 of 2.

_**Relates**: #125_

### CODE REVIEW

**Comment by lishaduck** [comment]
FYI, test failures are outdated snapshots, not #189.

File: new-package/elm-review-package-tests/check-previews-compile.js [hunk 1]
```diff
@@ -82,12 +82,13 @@ function checkDepsAreCompatible(exampleConfiguration, previewDependencies) {
...
     delete previewDependencies[depName];
-  });
+  }
 
   const remainingKeys = Object.keys(previewDependencies);
-  if (remainingKeys.length !== 0) {
+  if (remainingKeys.length > 0) {
     console.error(
-      `There are extraneous dependencies in the ${exampleConfiguration}/ configuration: ${remainingKeys}`
+      // prettier-ignore
```
Comment (jfmengels) [inline new-package/elm-review-package-tests/check-previews-compile.js:90]: I have to admit I'm not fond in having disable checks (for prettier or for TS) in files that package authors will then own.
Maybe in this case, we can create a new variable `const keys = remainingKeys.join(', ');` above `console.error(...)`?
That should get rid of the need for the ignore.

File: new-package/elm-review-package-tests/check-previews-compile.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
 #!/usr/bin/env node
 
-const path = require('path');
+const path = require('node:path');
 const Ansi = require('./helpers/ansi');
-const {execSync} = require('child_process');
+const {execSync} = require('node:child_process');
 const {findPreviewConfigurations} = require('./helpers/find-configurations');
+// @ts-ignore - Generated file.
```
Comment (lishaduck) [inline new-package/elm-review-package-tests/check-previews-compile.js:7]: Here's another ignore comment. I figure we can call `.replace()` on the file and delete this line.

Comment (lishaduck) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: Did it!

File: lib/new-package.js [hunk 3]
```diff
@@ -310,6 +310,14 @@ ElmjutsuDumMyM0DuL3.elm
     }
   );
 
+  const checkPreviewsCompile = path.join(
+    packageTests,
+    'check-previews-compile.js'
+  );
+  const data = await FS.readFile(checkPreviewsCompile);
+  const result = data.replace(/\/\/ @ts-ignore - Generated file.\n/, '');
+  await FS.writeFile(checkPreviewsCompile, result, 'utf8');
```
Comment (lishaduck) [inline lib/new-package.js:319]: I'm rather proud of myself: I wrote some JS that didn't blow up in my face 🤣

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 2 above)
Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: Not optimal in terms of performance, but I guess it works :sweat_smile:

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 1 above)
Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:90]: You marked this as resolved, but the comment is still here. Did you apply it on a different branch maybe?

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 2 above)
Comment (lishaduck) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: I figure that if we go the route of hardcoding paths for #147, then this wouldn't be so bad 🤷‍♂️
Then again, that's not a performance optimal route...

Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: Yes, not an issue in terms of performance. But I'm not sure I get what the relation with #147 is, sorry

Comment (lishaduck) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: I thought you meant that we were copying the files, then editing them, but with #147, we'll move them one by one, so we could cut out the duplication, but at the expense of total speed (as a native cp is certainly faster than our kinda-pure-js one would be.)

Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: That is what I meant. Even with #147 and specifying paths we still would do this copy+edit+write process for this file, no?

Actually, can't we add a dummy `new-package/elm.json` file to get rid of the TS error entirely?

Comment (lishaduck) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: Yeah, I guess we're just talking past each other.
We could. Would our actual one overwrite it?

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 1 above)
Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:90]: Let's resolve this later, let's not block all the other PRs because of this.

**Comment by jfmengels** [comment]
I'll merge this PR. We can always improve things later, I don't want you to fall into Git branch hell :sweat_smile: 
And thanks again for all this work, it's incredible! :heart:

**Comment by lishaduck** [comment]
> I'll merge this PR. We can always improve things later, I don't want you to fall into Git branch hell 😅 And thanks again for all this work, it's incredible! ❤️

It's a bit too late for that 🤣:
Screenshot 2024-07-08 at 5 10 26 PM

I mean, that's what summer break is for, eh? Finding projects? Enjoying not having to tune out teachers while you edit this and that? 😉

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 2 above)
Comment (jfmengels) [reply new-package/elm-review-package-tests/check-previews-compile.js:7]: I think that if we get in #147 then we can skip copying the dummy `elm.json`, so maybe it should happen in tandem with #147 (which btw, feel free to take over, I don't remember the state of it).

**Comment by jfmengels** [comment]
Right, but if I waited a few more days, who knows how much worse it would have been :smile:

**Comment by lishaduck** [comment]
> Right, but if I waited a few more days, who knows how much worse it would have been 😄

True, true... YIKES!

File: new-package/elm-review-package-tests/check-previews-compile.js (hunk 1 above)
Comment (lishaduck) [reply new-package/elm-review-package-tests/check-previews-compile.js:90]: Huh. Well, once it's all merged I can check again.

**Pull request merged**